x-pack/plugins/security_solution/common/search_strategy/security_solution/index.ts

```diff
@@ -40,7 +40,7 @@ export enum Direction {
 }
 
 export interface SortField {
-  field: string;
+  field: 'lastSeen' | 'hostName';
```

Comment:
NOTE: I think the assert type was trying to hit here that this is not typed and it should be typed or the unreachable assert should be removed as it cannot be an exhaustive switch if it's on just a string.

Reply:
/cc @XavierM

Reply:
it should be generic since when we create all hosts query it was typed but all the other queries after that with sort field have not been typed. So I imagine that we want it to be consistent and just a string to keep it generic since all the other queries after all hosts have never been with safe type on the sort field.

Reply:
Also slept on this one and since we are not using `graphql` for our type anymore, and I think we can be stricter and more generic. So please @FrankHassanabad keeps it the way it is, at the same time that we are refactoring our graphql queries to search strategy. I think that I will do something like that to be more generic for every query who needs sorting. Do you think that something like that will make sense?
```
export interface SortField<Field = string> {
  field: Field;
...
```

Reply:
Yeah that should work out. Nothing wrong with being generic IMO when we need to be generic. It's going to be the non-ending balance and strife between the finer points of when dynamic code and types vs static code and types.

We will forever be caught up in these. I accidentally broke type safety within actions in 7.9.0 (for example) with types and meta during refactoring last release and then that caused me to introduce a bug in 7.9.0 with actions and alerts I scrambled late yesterday to fix.

Which is why I can probably see these at the moment. Thanks for the code review and notes.